pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -375,6 +392,18 @@ func (i *instantSplitter) splitAndSquashCall(expr *parser.Call, stats *MapperSta
 	return squashExpr, true, nil
 }
 
+// assertRangeInterval returns the range interval specified in the input expr and whether it is greater than
+// the configured split interval.
+func (i *instantSplitter) assertRangeInterval(expr parser.Expr) (rangeInterval time.Duration, canSplit bool) {
```

Comment:
Nit: 
```suggestion
func (i *instantSplitter) assertSplittableRangeInterval(expr parser.Expr) (rangeInterval time.Duration, canSplit bool) {
```
Since the `getRangeInterval` also discards binary expressions and subqueries